replica/table.cc

```diff
@@ -1121,7 +1121,6 @@ table::try_flush_memtable_to_sstable(compaction_group& cg, lw_shared_ptr<memtabl
           try {
             sstables::sstable_writer_config cfg = get_sstables_manager().configure_writer("memtable");
             cfg.backup = incremental_backups_enabled();
-            cfg.erm = _erm;
```

Comment:
" - bad: holding on to an effective_replication_map_ptr is a barrier
   for topology operations, preventing tablet migrations (etc) while
   an sstable is being written"

it may not be a problem as we synchronize anyway by flushing before streaming on migration, right? newer writes, during flushing, reaches the pending replica too. sstable writes of compaction is not a problem.

Reply:
Unrelated compactions (different tablets, different tables, different nodes) also prevent the topology state machine from advancing.